From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Strokkur24 <strokkur.24@gmail.com>
Date: Wed, 8 Apr 2026 02:30:43 +0200
Subject: [PATCH] Add plugin loader utility methods


diff --git a/src/main/java/io/papermc/paper/plugin/loader/library/impl/MavenLibraryResolver.java b/src/main/java/io/papermc/paper/plugin/loader/library/impl/MavenLibraryResolver.java
index 298bb04c294a89ad783eb3dac73faf47fa5fb1ba..57e61fe33f86db5af3ed02477817d84154e4cef1 100644
--- a/src/main/java/io/papermc/paper/plugin/loader/library/impl/MavenLibraryResolver.java
+++ b/src/main/java/io/papermc/paper/plugin/loader/library/impl/MavenLibraryResolver.java
@@ -128,6 +128,29 @@ public class MavenLibraryResolver implements ClassPathLibrary {
         }
         this.repositories.add(remoteRepository);
     }
+    // Tabletop start - add plugin loader utility methods
+
+    /// Adds the provided dependency to the library resolver.
+    /// This is a utility method for [#addDependency(org.eclipse.aether.graph.Dependency)].
+    ///
+    /// @param coords A Gradle-style dependency string: `com.example:artifact:1.0.0-SNAPSHOT`
+    /// @see #addDependency(Dependency)
+    public void addDependencyCoords(final String coords) {
+        this.addDependency(new Dependency(new org.eclipse.aether.artifact.DefaultArtifact(coords), null));
+    }
+
+    /// Adds the provided repository to the library resolver.
+    /// This is a utility method for [#addRepository(RemoteRepository)].
+    ///
+    /// This method sets the `id` to `null` and `type` of the repository to `default`. If you need
+    /// more fine-grained control over the repository, you should prefer [#addRepository(RemoteRepository)].
+    ///
+    /// @param url the URL to the repository
+    /// @see #addRepository(RemoteRepository)
+    public void addRepositoryUrl(final String url) {
+        this.addRepository(new RemoteRepository.Builder(null, "default", url).build());
+    }
+    // Tabletop end
 
     /**
      * Resolves the provided dependencies and adds them to the library store.
